### ISSUE TITLE
Issues in `spk install-build-pipeline` produced azure-pipelines.yaml file

### ISSUE THREAD

**Issue by andrebriggs** [microsoft/bedrock#907]
**Describe the bug:**
I was able to get pass #906 issue by manually changing the AzDO pipeline path to the `azure-pipelines.yaml` file. Once that is done I can run a pipeline scaffolded by the `spk service install-build-pipeline` command. 

The problem is that the `azure-pipelines.yaml` file doesn't seem to handle non-mono repos. A concatenation error makes the Docker image name invalid.

Below you will see that `fabrikam-frontend-:master-20200126.1` is not a valid Docker name image. 

<pre>
/bin/bash --noprofile --norc /home/vsts/work/_temp/6d5772c7-942c-41c9-99de-ce19c9531f73.sh
<b>Image Name: fabrikam-frontend-</b>
az acr build -r andrebriggs --image <b>fabrikam-frontend-:master-20200126.1 .</b>
WARNING: Packing source code into tar to upload...
WARNING: Excluding '.gitignore' based on default ignore rules
WARNING: Excluding '.git' based on default ignore rules
WARNING: Uploading archived source code from '/tmp/build_archive_f15dc90c825941dfaf2161e7c06e8e28.tar.gz'...
WARNING: Sending context (2.010 KiB) to registry: andrebriggs...
<b>ERROR: Some of the properties of 'runRequest' are invalid.. InnerErrors: DockerBuildRequest:Image name 'fabrikam-frontend-:master-20200126.1' is invalid. 
The acceptable patterns are 'repository' or 'repository:tag'.The repository and tag names follow the standardized docker repository and tag naming conventions. Tag templates are supported within '{{' and '}}'. </b>
Please refer to docs for more information on using image name and tag:  https://aka.ms/az/acr/az-acr-build
##[error]Bash exited with code '1'.
Finishing: ACR Build and Publish
</pre>

The build worked once I removed the `-`. See below:

![image](https://user-images.githubusercontent.com/3119513/73129263-1e0b7b00-3f94-11ea-8b6b-5fdabc05bc68.png)

I assume that there was supposed to be something injected into the script after the `-`. Maybe that didn't happen in the non-mono repo situation

Another issue is the creation of a variable named `PROJECT_NAME_LOWER` in the `azure-pipelines.yaml` generated by the `spk service install-build-pipeline` command. Someone `PROJECT_NAME_LOWER`doesn't get populated and that results in a PR on a branch with two dashes. Example: 

![image](https://user-images.githubusercontent.com/3119513/73129306-4051c880-3f95-11ea-8ab3-e77731c7313c.png)

```Merge DEPLOY/fabrikam-frontend--master-20200126.2 to refs/heads/master```

![image](https://user-images.githubusercontent.com/3119513/73129291-e8b35d00-3f94-11ea-81a0-ec9b15e0f3f2.png)

I fixed this by removing the bolded part below. This is certainly a hack

<pre>
 - script: |-
              export PROJECT_NAME_LOWER=$(echo  | tr '[:upper:]' '[:lower:]')
              export BUILD_REPO_NAME=$(echo $(Build.Repository.Name)<b>-$PROJECT_NAME_LOWER</b> | tr '[:upper:]' '[:lower:]')
              # --- From https://raw.githubusercontent.com/Microsoft/bedrock/master/gitops/azure-devops/release.sh
              . build.sh --source-only
</pre>

**To Reproduce:**

See #906 

**Expected behavior:**
Build from `azure-pipelines.yaml` should work and produce the correct PR on the HLD repo

**Additional context:**

**Does this require updates to documentation?:**

**Comment by mtarng** [comment]
Related: https://github.com/microsoft/bedrock/issues/801

Refactoring for the build pipeline, since we're only generating one pipeline file, and that pipeline file should only do builds for a single service

**Comment by andrebriggs** [comment]
@mtarng as of v0.5.0 of SPK this issue of having a dash at the end of the BUILD_REPO_NAME has been expanding into the Spektate task of the install-build-pipeline
![image](https://user-images.githubusercontent.com/3119513/73692376-09f90500-4689-11ea-8f59-c411ee7e6eb3.png)
